### ISSUE TITLE
Avoid :verify option to get overridden

### ISSUE THREAD

**Issue by guapolo** [jnunemaker/httparty#266]
Hi
`attach_ssl_certificates` overrides the `:verify` option value in this line https://github.com/jnunemaker/httparty/blob/master/lib/httparty/connection_adapter.rb#L147

IMHO this should not be the case, using `:pem` or `pfx`, this options should honor the previously set `:verify_mode` value.

I have a use case where using `:pem` certificates and `:verify` set to `false`   (in my case, I'm testing my code against a web service which requieres pem certificates for authentication, but the server doesn't have a signed certificate in it's testing environment).

In other words, here's the code I was using, and expected it to work:

```ruby
class MyClass
  include HTTParty
  default_params(verify: false, pem: a_certificate_and_key)
 # ...
 post(url, ...) # when posting, verify_mode is set to OpenSSL::SSL::VERIFY_PEER
end
```

but instead I need to use this declaration for it to work `OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE`, which is pretty ugly.

I'll submit a pull request if you agree on this behavior.

Thanks,

**Comment by kurenn** [comment]
I just placed the pull request https://github.com/jnunemaker/httparty/pull/441

Had some issues with this, when working with a client, it is working now!

**Comment by kurenn** [comment]
Solution has been merged with https://github.com/jnunemaker/httparty/pull/441

@jnunemaker I think you can close this issue
